crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -254,6 +254,30 @@ pub fn impl_param_set(_input: TokenStream) -> TokenStream {
                 fn apply(&mut self, world: &mut World) {
                     self.0.apply(world)
                 }
+
+                fn world_access_level() -> WorldAccessLevel {
+                    let mut exclusive = false;
```

Comment:
Nit: I feel like this could be done more reliably / clearly via an `Ord` impl and a reduce.